### PR TITLE
samples: foc_controller: replace sine modulation

### PR DESCRIPTION
**Pull request by uLipe**
with space vector pwm generation, also fixes the rotor
alignment procedure and a problem with step processor
manager that creates a dead loop thread when poll
rate is zero but event driven thread is not defined.

Signed-off-by: Felipe Neves <felipe.neves@linaro.org>

### CODE REVIEW

**Comment by uLipe** [comment]
@microbuilder PTAL, this needs to be added befoere sensorless version gets in place.

File: samples/foc_controller/src/foc_driver.c [hunk 1]
```diff
@@ -220,37 +186,38 @@ int foc_driver_initialize(float motor_pole_pairs, float dc_link_voltage, foc_mea
...
-	foc_driver_set_voltages(0.1f * bus_dc_link_voltage, 0.0f);
-	k_sleep(K_MSEC(500));
-
 	encoder_reading.zero_offset = foc_driver_read_encoder() & AS5600_READING_MASK;
 
 	return 0;
 }
 
-void foc_driver_set_voltages(float voltage_u, float voltage_v)
+void foc_driver_set_duty_cicle(float dc_u, float dc_v, float dc_w)
```
Comment (microbuilder) [inline samples/foc_controller/src/foc_driver.c:194]:
```suggestion
void foc_driver_set_duty_cycle(float dc_u, float dc_v, float dc_w)
```

File: samples/foc_controller/src/foc_driver.h [hunk 2]
```diff
@@ -73,14 +71,17 @@ int foc_driver_initialize(float motor_pole_pairs, float dc_link_voltage, foc_mea
...
+ *
+ * @param dc_v voltage to be applied to the in phase v of motor
  *
- * @param voltage_v voltage to be applied to the in phase v of motor
+ * @param dc_v voltage to be applied to the in phase v of motor
  * 
+ * @param values should be normalized between 0 and 1 
  */
-void foc_driver_set_voltages(float voltage_u, float voltage_v);
+void foc_driver_set_duty_cicle(float dc_u, float dc_v, float dc_w);
```
Comment (microbuilder) [inline samples/foc_controller/src/foc_driver.h:84]:
```suggestion
void foc_driver_set_duty_cycle(float dc_u, float dc_v, float dc_w);
```

File: samples/foc_controller/src/foc_driver.h [hunk 3]
```diff
@@ -73,14 +71,17 @@ int foc_driver_initialize(float motor_pole_pairs, float dc_link_voltage, foc_mea
 int foc_driver_set_encoder_offset(void);
 
 /**
- * @brief set inverter voltages.
+ * @brief set inverter voltages based on duty cicle.
```
Comment (microbuilder) [inline samples/foc_controller/src/foc_driver.h:74]:
```suggestion
 * @brief set inverter voltages based on duty cycle.
```

File: samples/foc_controller/src/foc_nodes.c [hunk 4]
```diff
@@ -73,23 +111,26 @@ int foc_do_process(struct step_measurement *mes, uint32_t handle, uint32_t inst)
...
-						p->cmd.voltage_v);
+	/* convert the alpha-beta frame into SVPWM signals */
+	foc_svm_set(p->cmd.voltage_alpha, 
+			p->cmd.voltage_beta,
+			&inverter_duties[0],
+			&inverter_duties[1],
+			&inverter_duties[2]);
 
+	/* and transfer them directly to the inverter */
+	foc_driver_set_duty_cicle(inverter_duties[0], inverter_duties[1], inverter_duties[2]);
```
Comment (microbuilder) [inline samples/foc_controller/src/foc_nodes.c:132]:
```suggestion
	foc_driver_set_duty_cycle(inverter_duties[0], inverter_duties[1], inverter_duties[2]);
```

File: samples/foc_controller/src/foc_svm.h [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2022 Linaro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief set inverter voltages based on duty cicle.
```
Comment (microbuilder) [inline samples/foc_controller/src/foc_svm.h:8]:
```suggestion
 * @brief set inverter voltages based on duty cycle.
```

File: samples/foc_controller/src/main.c [hunk 6]
```diff
@@ -72,8 +70,8 @@ static int step_foc_cmd_align(const struct shell *shell, size_t argc, char **arg
 /* Subcommand array for "step_foc" (level 1). */
 SHELL_STATIC_SUBCMD_SET_CREATE(
 	sub_step_foc,
-	SHELL_CMD(set, NULL, "Set voltages to control the motor", step_foc_cmd_set),
-	SHELL_CMD(align, NULL, "Align rotor and calibrate sensor", step_foc_cmd_align),	
+	SHELL_CMD(set, NULL, "Set voltage d-q vector to control the motor between -1 and +1", step_foc_cmd_set),
+	SHELL_CMD(align, NULL, "Align rotor and calibrate its position sensor", step_foc_cmd_align),
```
Comment (microbuilder) [inline samples/foc_controller/src/main.c:74]:
```suggestion
	SHELL_CMD(align, NULL, "Align rotor and calibrate it’s position sensor", step_foc_cmd_align),	
```

**Review by microbuilder — CHANGES_REQUESTED** [review]

**Comment by uLipe** [comment]
@microbuilder PTAL again.

**Comment by uLipe** [comment]
@microbuilder PTAL

**Review by microbuilder — APPROVED** [review]

**Pull request merged**